consensus/consensus-types/src/vote.rs

```diff
@@ -62,6 +62,16 @@ impl Vote {
         }
     }
 
+    /// Returns true if the two votes are equivalent. This does not take into account the timeout signature.
+    pub fn equivalent_to(&self, other: &Self) -> bool {
+        let mut equivalent = true;
```

Comment:
I think you can just compare ledger_info.hash, `verify` already guarantees other fields hash matches consensus_data_hash inside

Reply:
OK I see, the only field that can still differ is the signature but this needs to be tolerated because of malleability

![Screen Shot 2020-07-07 at 8 33 23 PM](https://user-images.githubusercontent.com/1316043/86872397-1bf42f00-c091-11ea-9ecc-e966716fa8c4.png)